### PR TITLE
Add simplenameversion watch struct and interfaces from local regis…

### LINKED ISSUE #3: Add simple dogu version struct to domain objects

### PR DESCRIPTION
**Pull request by nhinze23**
…try usage.

Resolves #3

### CODE REVIEW

File: dogu/interfaces.go [hunk 1]
```diff
@@ -17,3 +17,19 @@ type RemoteDoguDescriptorRepository interface {
...
+	WatchAllCurrent(context.Context) (<-chan CurrentVersionsWatchResult, error)
+}
+
+// LocalDoguDescriptorRepository is an append-only Repository, no updates will happen
+type LocalDoguDescriptorRepository interface {
+	Get(context.Context, SimpleNameVersion) (*core.Dogu, error)
+	GetAll(context.Context, []SimpleNameVersion) (map[SimpleNameVersion]*core.Dogu, error)
+	Add(context.Context, SimpleName, *core.Dogu) error
+	DeleteAll(context.Context, SimpleName) error
+}
```
Comment (alexander-dammeier) [inline dogu/interfaces.go]: Do we want to add docs here? At the moment this says nothing about explicit error types for different cases etc.

File: dogu/doguName.go [hunk 2]
```diff
@@ -15,6 +15,18 @@ func (s SimpleName) String() string {
...
+	Version core.Version
+}
+
+type CurrentVersionsWatchResult struct {
+	Versions     map[SimpleName]core.Version
+	PrevVersions map[SimpleName]core.Version
+	Diff         []SimpleNameVersion
+	Err          error
+}
+
```
Comment (alexander-dammeier) [inline dogu/doguName.go]: Please add go doc

File: dogu/interfaces.go [hunk 3]
```diff
@@ -17,3 +17,19 @@ type RemoteDoguDescriptorRepository interface {
...
+	GetCurrent(context.Context, SimpleName) (SimpleNameVersion, error)
+	GetCurrentOfAll(context.Context) ([]SimpleNameVersion, error)
+	IsEnabled(context.Context, SimpleNameVersion) (bool, error)
+	Enable(context.Context, SimpleNameVersion) error
+	WatchAllCurrent(context.Context) (<-chan CurrentVersionsWatchResult, error)
+}
+
+// LocalDoguDescriptorRepository is an append-only Repository, no updates will happen
+type LocalDoguDescriptorRepository interface {
+	Get(context.Context, SimpleNameVersion) (*core.Dogu, error)
```
Comment (alexander-dammeier) [inline dogu/interfaces.go:51]: Do we already have a dependency to the cesapp-lib, so it is no new dependency with `*core.Dogu` now?

**Review by alexander-dammeier — CHANGES_REQUESTED** [review]
I thin k we need a little bit of documentation for it. Maybe you can copy go doc from the previous implementation of the types and interfaces.

File: dogu/interfaces.go (hunk 3 above)
Comment (nhinze23) [reply dogu/interfaces.go:51]: Version is in cesapp-lib too

File: dogu/interfaces.go [hunk 4]
```diff
@@ -17,3 +17,48 @@ type RemoteDoguDescriptorRepository interface {
...
 	Get(context.Context, QualifiedVersion) (*core.Dogu, error)
 }
+
+type VersionRegistry interface {
+	// GetCurrent returns the current installed dogu version.
+	// NotFound Error if the dogu is not installed.
+	// ConnectionError if there are any connection issues.
+	// Generic Error if there are any other issues.
+	GetCurrent(context.Context, SimpleName) (SimpleNameVersion, error)
+	// GetCurrentOfAll returns all current installed dogu version.
```
Comment (alexander-dammeier) [inline dogu/interfaces.go]:
```suggestion
	// GetCurrentOfAll returns all current installed dogu versions.
```

File: dogu/interfaces.go [hunk 5]
```diff
@@ -17,3 +17,48 @@ type RemoteDoguDescriptorRepository interface {
...
+	Get(context.Context, SimpleNameVersion) (*core.Dogu, error)
+	// GetAll returns all dogu descriptor by parameter from the local registry.
+	// ConnectionError if there are any connection issues
+	// Generic Error if there are any other issues.
+	GetAll(context.Context, []SimpleNameVersion) (map[SimpleNameVersion]*core.Dogu, error)
+	// Add adds a dogu descriptor to the local registry.
+	// ConnectionError if there are any connection issues
+	// Generic Error if there are any other issues.
+	Add(context.Context, SimpleName, *core.Dogu) error
+	// DeleteAll deletes all dogu descriptors by name.
```
Comment (alexander-dammeier) [inline dogu/interfaces.go]:
```suggestion
	// DeleteAll deletes all dogu descriptors for a given dogu name.
```

**Review by alexander-dammeier — APPROVED** [review]
Just two quickfixes. The rest looks good to me.

**Pull request merged**